autoconnect/autoconnect-ws/autoconnect-ws-sm/src/identified/on_client_msg.rs

```diff
@@ -232,8 +231,21 @@ impl WebPushClient {
                         .db
                         .remove_message(&self.uaid, &n.chidmessageid())
                         .await?;
+                    // NOTE: timestamp messages may still be in state of flux: they're not fully
```

Comment:
We should really only mark this as `Delivered` if the `ACK` code is [100](https://searchfox.org/mozilla-central/source/dom/push/PushServiceWebSocket.sys.mjs#27). The others aren't really 'ACK`s and probably should be considered Errors.

Reply:
How about separate states for these instead of Errors (especially since we have always treated them as normal ACKs)? 100: Delivered, 101: DecryptionError, 102: NotDelivered

Reply:
I think I'm ok with that. I guess I was more thinking that we'd only really consider a message as "Delivered" if the UA returned a code of 100, since that means that the webapp actually received the message. We could consider receipt by the User Agent to be "Delivered" but we should explain why the webapp might not have processed it correctly (a bit out of scope for us, but whatever)